### PR TITLE
Docker workflow improvement

### PR DESCRIPTION
**Pull request by ilteoood**
Hi,
with this PR I would like to improve the docker workflow.
I would like to use alpine to make a smaller image, and a .dockerignore file to speed up the creation of the image in the local environment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-FROM node:16-buster-slim
-RUN apt update && apt upgrade -y && apt install ffmpeg git -y
+FROM node:16-alpine
+RUN apk update && apk upgrade && apk add ffmpeg git && rm -rf /var/cache/apk/*
```
Comment (rojserbest) [inline Dockerfile:2]: What's that "`rm -rf /var/cache/apk/`" for?

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ node_modules/
 dist/
 cmr*
 .env
+.vscode/
```
Comment (rojserbest) [inline .gitignore:5]: Please remove `.vscode` as it might be added by us later.

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,14 @@
+node_modules
+.github
+dist
+.dockerignore
+.eslintrc.yml
+.gitignore
+.prettierrc
+CODE_OF_CONDUCT.md
+CONTRIBUTING.md
```
Comment (rojserbest) [inline .dockerignore:9]: Can you please add `.vscode` and replace all `.md` files with `*.md`?

**Review by rojserbest — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (ilteoood) [reply Dockerfile:2]: Is used to clean the internal cache of alpine, in order to keep the image smaller.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilteoood) [reply .gitignore:5]: I would like to contribute more to this project.
In my `.vscode` folder I have the local history of the edited files. I would like to keep it.

[the remark this answers is not in the document]
Comment (rojserbest) [reply .gitignore:5]: Maybe do that in a different folder. How about `dist/`? :)

**Comment by rojserbest** [comment]
You should also update the `devDependencies` in the `package-lock.json`. Not manually, but let `npm` do that.

File: .gitignore (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilteoood) [reply .gitignore:5]: The settings is for workspace, not for project.
Dist folder is cleared after each build, while I need them for the entire development.

Why is this a problem? This folder should never be committed.

**Comment by ilteoood** [comment]
I opened another PR, as I rebased the branch ant this PR isn't mergeable anymore.

**Comment by rojserbest** [comment]
Why added the example of what not to do?

Because you don't have to open more PRs if you did a mistake in one. You can just forget your mistake and add more commits correcting your mistake.

**Pull request closed without merging**